File: app/src/main/java/com/mbestavros/geometricweather/background/receiver/widget/AbstractWidgetProvider.java
package com.mbestavros.geometricweather.background.receiver.widget;

import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.Context;

import com.mbestavros.geometricweather.utils.helpter.IntentHelper;

/**
 * Abstract widget provider.
 * */
public class AbstractWidgetProvider extends AppWidgetProvider {

    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        super.onUpdate(context, appWidgetManager, appWidgetIds);
        IntentHelper.startAwakeForegroundUpdateService(context);
    }
}

File: app/src/main/java/org/breezyweather/background/receiver/widget/AbstractWidgetProvider.kt
package org.breezyweather.background.receiver.widget

import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.Context
import org.breezyweather.remoteviews.Widgets

/**
 * Abstract widget provider.
 */
abstract class AbstractWidgetProvider : AppWidgetProvider() {
    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
        super.onUpdate(context, appWidgetManager, appWidgetIds)
        Widgets.updateWidgetIfNecessary(context)
    }
}
